app/src/home.js

```diff
@@ -171,28 +173,55 @@ export default function Home() {
   const [HealthCareProfessionals, setHealthCareProfessionals] = useState([]);
   const [healthCareProfessionalIdDisabled] = useState(false);
   const [patientIdDisabled] = useState(false);
+  const [comprehendResults, setComprehendResults] = useComprehension(transcripts || [], transcribeCredential);
+  const [soapSummary, setSOAPSummary] = useState(() => generateSOAPSummary([].concat(...comprehendResults)));
+
+  useEffect(() => {
+    if (!showSOAPReview) {
+      setSOAPSummary(generateSOAPSummary([].concat(...[...comprehendResults, ...comprehendCustomEntities])));
+    }
+  }, [comprehendResults, comprehendCustomEntities, showSOAPReview]);
+
+  function updateSOAPSummary(e) {
+    setSOAPSummary(e.target.value);
+  }
 
   var sessionId = '';
 
   const history = useHistory();
 
+  const onTranscriptChange = (i, value) => {
+    setTranscripts((t) => {
+      if (t[i].text === value) {
+        return t;
+      }
+      const newChunk = {
+        ...t[i],
+        text: value,
+      };
+
+      return [...t.slice(0, i), newChunk, ...t.slice(i + 1)];
+    });
+  };
+
   const addTranscriptChunk = useCallback(({ Alternatives, IsPartial, StartTime }) => {
-    const [text] = Alternatives[0].Items.reduce(
-      ([prevText, prevAddSpeakerLabel], item) => {
-        const isSpeakerChange = item.Type === 'speaker-change';
-        const shouldAddSpeakerLabel = !isSpeakerChange && prevAddSpeakerLabel && 'Speaker' in item;
-        const isPronunciation = item.Type === 'pronunciation';
-        const isPunctuation = item.Type === 'punctuation';
-        const speakerLabel = shouldAddSpeakerLabel ? `Speaker ${(parseInt(item.Speaker) + 1).toString()}\n` : '';
-        const itemContent = isPronunciation || isPunctuation ? item.Content : '';
-        const spaceAtEnd = isPronunciation ? ' ' : '';
-        const text = `${prevText}${speakerLabel}${itemContent}${spaceAtEnd}`;
-        const addSpeakerLabel = isSpeakerChange || (shouldAddSpeakerLabel ? false : prevAddSpeakerLabel);
-
-        return [text, addSpeakerLabel];
-      },
-      ['', true],
-    );
+    let addSpeakerLabel = true;
```

Comment:
This entire codeblock needs to the one that is on master. Please revert thie addTranscriptChunk to what is on master

Reply:
Reverted. Will push a new change